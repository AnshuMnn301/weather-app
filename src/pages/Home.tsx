import React from 'react';
import {Button, TextInput, View} from 'react-native';

export default function Home({navigation}: any) {
  const [text, OnChangeText] = React.useState('');
  return (
    <View
      style={{
        flexDirection: 'row',
        alignItems: 'center',
        flex: 1,
        justifyContent: 'center',
      }}>
      <TextInput
        placeholder="Enter country name"
        style={{
          width: '60%',
          borderBottomColor: 'lightgray',
          borderBottomWidth: 1,
          marginRight: 10,
          fontSize: 20,
        }}
        value={text}
        onChangeText={OnChangeText}
      />
      <Button
        title="Search"
        disabled={text === ''}
        onPress={() =>
          navigation.navigate('Details', {
            country: text,
          })
        }
      />
    </View>
  );
}
