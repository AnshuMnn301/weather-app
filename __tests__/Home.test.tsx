/**
 * @format
 */
import {Button} from 'react-native'
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Home from '../src/pages/Home';

test('render Home Component corrcetly', () => {
    const tree = renderer.create(<Home navigation={{
        navigate: jest.fn()
    }}/>).toJSON()
    expect(tree).toMatchSnapshot()
})
describe('Testing Home components', () => {
    test('Test 1', async () => {
        const navigateMockFn = jest.fn()
        const wrapper = shallow(<Home navigation={{
            navigate: navigateMockFn
        }}/>)
        const button = wrapper.find(Button)
        button.props().onPress()
        wrapper.update()
        expect(navigateMockFn).toHaveBeenCalledWith('Details', {"country": ""})
    })
})


