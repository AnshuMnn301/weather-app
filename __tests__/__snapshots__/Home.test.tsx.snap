// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Home Component corrcetly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    placeholder="Enter country name"
    style={
      Object {
        "borderBottomColor": "lightgray",
        "borderBottomWidth": 1,
        "fontSize": 20,
        "marginRight": 10,
        "width": "60%",
      }
    }
    value=""
  />
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {},
        ]
      }
    >
      <Text
        disabled={true}
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#cdcdcd",
            },
          ]
        }
      >
        Search
      </Text>
    </View>
  </View>
</View>
`;
